app/src/main/java/org/schabi/newpipe/info_list/dialog/StreamDialogDefaultEntry.java

```diff
@@ -110,6 +116,35 @@ public enum StreamDialogDefaultEntry {
             ShareUtils.shareText(fragment.requireContext(), item.getName(), item.getUrl(),
                     item.getThumbnailUrl())),
 
+    DOWNLOAD(R.string.download, (fragment, item) -> ExtractorHelper.getStreamInfo(
+                    item.getServiceId(),
+                    item.getUrl(),
+                    false
+            )
+            .subscribeOn(Schedulers.io())
+            .subscribe((info) -> {
+                        final ArrayList<VideoStream> sortedVideoStreamsList = new ArrayList<>(
+                                ListHelper.getSortedStreamVideosList(fragment.requireContext(),
+                                        info.getVideoStreams(), null, false, false));
+                        final int index = ListHelper.getDefaultResolutionIndex(
+                                fragment.requireContext(), sortedVideoStreamsList);
+
+                        final DownloadDialog downloadDialog =
+                                DownloadDialog.newInstance(info);
+                        downloadDialog.setVideoStreams(sortedVideoStreamsList);
+                        downloadDialog.setSelectedAudioStream(index);
+                        downloadDialog.setAudioStreams(info.getAudioStreams());
+                        downloadDialog.setSubtitleStreams(info.getSubtitles());
```

Comment:
This piece can be removed if you just use the following instead:
```java
final DownloadDialog downloadDialog = DownloadDialog.newInstance(fragment.requireContext(), info);
```

While you are at it, may I ask you to also replace [these lines in RouterActivity](https://github.com/TeamNewPipe/NewPipe/blob/1e2d76a686b095a65f8052f68e4bbc12625a2108/app/src/main/java/org/schabi/newpipe/RouterActivity.java#L637-L648) with:
```java
final FragmentManager fm = getSupportFragmentManager();
final DownloadDialog downloadDialog = DownloadDialog.newInstance(this, result);
```